gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/ApiPlansResource.java

```diff
@@ -236,7 +239,11 @@ public Response closeApiPlan(
                     .build();
         }
 
-        return Response.ok(planService.close(plan, getAuthenticatedUser())).build();
+        PlanEntity closedPlan = planService.close(plan, getAuthenticatedUser());
+
+        removePlanFromApiDefinition(plan, api);
```

Comment:
it's not the right way to fix this.
For example a deprecated api cannot be added to a category. we have the same error.

Perhaps I'm missing something but why fix the close action and not the definition of a deprecated API ?

Reply:
@NicolasGeraud, I did not see the bug for adding to a category, which is an already existing bug.
When updating an API, we are checking if it is `DEPRECATED` (in this case there is an exception)  

The definition is updated after the plan is deleted / deprecated and not in PLanService which is itself called by ApiService.update()